proposals/text/3959-json-converter.md

```diff
@@ -0,0 +1,143 @@
+- Title: Addition of a `JsonConverter` node
+- Decision driver: @bglearning
+- Start Date: 2023-01-26
+- Proposal PR: #3959
+
+# Summary
+
+Right now we don't have a node that can take json files as input to be fed into a pipeline.
+
+Proposal: Add a `JsonConverter` node that takes in a json file, parses it, and generates `Document`s.
+It would also support the `jsonl` format with one line corresponding to one document.
+
+# Basic example
+
+```python
+from haystack.nodes import JsonConverter
+
+converter = JsonConverter()
+
+# Receive back List[Document]
+docs = converter.convert("data_file.json")
+```
+
+With the `data_file.json` as a list of json representation of documents:
+
+```json
+[
+    {
+        "content": "...",
+        "content_type": "text", "meta": {...}
+    },
+    {
+        "content": [["h1", "h2"], ["val1", "val2"]],
+        "content_type": "table", "meta": {...}
+    }
+]
+```
+
+Alternatively, the data can also be `jsonl`.
+By default, the converter will try to auto-detect between `json` and `jsonl`.
+Or the format can be explicitly specified during initialization with e.g. `JsonConverter(format='jsonl')`
```

Comment:
Minor, but do you think this brings any benefits in practice? The node should be able to tell with full confidence whether the file is JSONL or JSON. I don't see in which situations you'd like to specify `format`, as it only makes the node more likely to crash.

Reply:
Yes, true. The `format` specifier isn't truly necessary. Will remove it.